src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -365,14 +377,15 @@ static SegmentAllocator slicingAllocator(MemorySegment segment) {
      * Equivalent to (but likely more efficient than) the following code:
      * {@snippet lang=java :
      * MemorySegment segment = ...
-     * SegmentAllocator prefixAllocator = (size, align) -> segment.asSlice(0, size);
+     * SegmentAllocator prefixAllocator = (size, align) -> segment.asSlice(0, size, align);
      * }
-     * <p>
-     * This allocator can be useful to limit allocation requests in case a client
+     * The returned allocator throws {@link IndexOutOfBoundsException} when a slice of the provided
+     * segment with the requested size and alignment cannot be found.
+     *
+     * @apiNote A prefix allocator can be useful to limit allocation requests in case a client
      * knows that they have fully processed the contents of the allocated segment before the subsequent allocation request
      * takes place.
-     * <p>
-     * While the allocator returned by this method is <em>thread-safe</em>, concurrent access on the same recycling
+     * @implNote While a prefix allocator is <em>thread-safe</em>, concurrent access on the same recycling
```

Comment:
I think the term is used pretty much all over the javadoc - e.g. look for this preamble:
```
 * <a href="{@docRoot}/java.base/java/lang/doc-files/ValueBased.html">value-based</a>,
 * immutable and thread-safe
 ```

Reply:
Right, I guess I'm asking: can/should we do better? ;) It might be nice to define the term once and for all for the entire JDK, similar to how 'reachability' also has a central definition.

But, arguably, that seems like too much scope creep for this PR, so let's save it for a potential followup.

Reply:
I think there is something we could start off from: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/annotations/ThreadSafe.html

This could be made under a separate issue.

Reply:
https://bugs.openjdk.org/browse/JDK-8309342